staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go

```diff
@@ -178,3 +178,57 @@ func TestParseGroupKind(t *testing.T) {
 		})
 	}
 }
+
+func TestToAPIVersionAndKind(t *testing.T) {
+	tests := []struct {
+		input        GroupVersionKind
+		GroupVersion string
+		Kind         string
+	}{
+		{
+			input:        GroupVersionKind{Version: "V1", Kind: "pod"},
+			GroupVersion: "V1",
+			Kind:         "pod",
+		},
+		{
+			input:        GroupVersionKind{},
+			GroupVersion: "",
+			Kind:         "",
+		},
+	}
+	for i, test := range tests {
+		version, kind := test.input.ToAPIVersionAndKind()
+		if version != test.GroupVersion {
+			t.Errorf("%d: expected version: %#v, got: %#v", i, test.GroupVersion, version)
+		}
+		if kind != test.Kind {
+			t.Errorf("%d: expected kind: %#v, got: %#v", i, test.Kind, kind)
+		}
+	}
+}
+
+func TestBestMatch(t *testing.T) {
+	tests := []struct {
+		kinds   []GroupVersionKind
+		targets []GroupVersionKind
+		output  GroupVersionKind
+	}{
+		{
+			kinds:   []GroupVersionKind{{Version: "V1", Kind: "pod"}, {Version: "V2", Kind: "pod"}},
+			targets: []GroupVersionKind{{Version: "V1", Kind: "pod"}},
+			output:  GroupVersionKind{Version: "V1", Kind: "pod"},
+		},
+		{
+			kinds:   []GroupVersionKind{{Version: "V1", Kind: "pod"}, {Version: "V2", Kind: "pod"}},
+			targets: []GroupVersionKind{{Version: "V3", Kind: "pod"}, {Version: "V4", Kind: "pod"}},
+			output:  GroupVersionKind{Version: "V3", Kind: "pod"},
```

Comment:
consider adding a description ([example](https://github.com/kubernetes/kubernetes/blob/2f21cff49df7fd0cd5de059e42f6c38675e27e10/cluster/gce/gci/append_or_replace_prefixed_line_test.go#L30)). In this case it's falling back to the first target

Reply:
I have added it. Thanks for your review.

Reply:
please make use of the desc like [here](https://github.com/kubernetes/kubernetes/blob/2f21cff49df7fd0cd5de059e42f6c38675e27e10/cluster/gce/gci/append_or_replace_prefixed_line_test.go#L156)